src/script_chainer/utils/process_utils.py

```
@@ -36,6 +38,35 @@ def graceful_kill_popen(proc: subprocess.Popen, timeout: float = 3) -> None:
+    close_events = {0, 1, 2, 5, 6}
+
+    def _handler(ctrl_type: int) -> bool:
+        if ctrl_type in close_events:
+            on_close()
+            return True
+        return False
+
+    handler = handler_type(_handler)
+    try:
+        registered = bool(ctypes.windll.kernel32.SetConsoleCtrlHandler(handler, True))
+    except Exception:
```

Ruff (0.15.10)

[warning] Do not catch blind exception: `Exception`

(BLE001)